### DISCUSSION TITLE
behaviour of match rules cross 2 virtual services

### DISCUSSION THREAD

**Discussion by saud89** [istio/istio#45158] [Q&A]
Hi,
We have a scenario, where we want to have 2 virtual services, with 2 matches in corresponding virtual service:   
httproute in virtualservice1:   
```
- match:
       - uri:
           prefix: /
   route:
        - destination:
            host: service1
```   
httproute in virtualservice2:   
```
- match:
       - uri:
           prefix: /abc
   route:
        - destination:
            host: service2
```   
The routing works correctly, when a request is made with prefix /abc, call lands to service2 and other calls land to service1. It works as per expectation.    
However I came across this documentation: https://istio.io/latest/docs/ops/best-practices/traffic-management/#split-virtual-services  where it mentions following: `Although the order of evaluation for rules in any given source VirtualService will be retained, the cross-resource order is UNDEFINED`. I tried this on multiple setups and the routing works as expected, but because of the document there is a doubt, if we were lucky that it worked, and could it be a case where the matching rule might not work on cluster restart?
Also, can anyone guide on how to ensure prefix match to "/" should not overpower match to "/abc", when these rules are written in 2 virtual services?

Thanks!

**Comment by howardjohn** [comment]
Officially, it is undefined behavior that could change in theory. There is some special logic specifically to make a "match anything" occur last though.

This is, admittedly, not great.

In http://gateway-api.org/, this problem is fixed with well established precedence rules for merging, which Istio implements with HTTPRoute
